src/langchain/request-decorator.ts

```diff
@@ -0,0 +1,22 @@
+const openpmOrigin = 'https://openpm.ai'
+
+export function openpmRequestDecorator({ apiKey }: { apiKey?: string }) {
+  return (request: Request) => {
+    const url = new URL(request.url)
+
+    // Check origin
+    if (url.origin !== openpmOrigin) {
+      return request
+    }
+
+    if (!apiKey) {
+      apiKey = process.env.OPENPM_API_KEY
```

Comment:
small nit: this will crash if packaged with some js bundlers out there, safer to do `typeof process !== 'undefined' && process.env.OPENPM_API_KEY` (which then doesn't work with some other bundlers, but at least never crashes)

I've also found to work across all envs you need to have the option of passing in api key as an arg